streams/test-utils/src/test/java/org/apache/kafka/streams/TestInputTopicTest.java

```diff
@@ -0,0 +1,312 @@
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.test.TestRecord;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.apache.kafka.common.utils.Utils.mkProperties;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.junit.Assert.assertThrows;
+
+public class TestInputTopicTest {
+
+    private final static String INPUT_TOPIC = "input";
+    private final static String OUTPUT_TOPIC = "output1";
+    private final static String INPUT_TOPIC_MAP = OUTPUT_TOPIC;
+    private final static String OUTPUT_TOPIC_MAP = "output2";
+
+    private TopologyTestDriver testDriver;
+    private final Serde<String> stringSerde = new Serdes.StringSerde();
+    private final Serde<Long> longSerde = new Serdes.LongSerde();
+
+    private final Properties config = mkProperties(mkMap(
+            mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "TestInputTopicTest"),
+            mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234")
+    ));
+    private final Instant testBaseTime = Instant.parse("2019-06-01T10:00:00Z");
+
+    @Before
+    public void setup() {
+        final StreamsBuilder builder = new StreamsBuilder();
+        //Create Actual Stream Processing pipeline
+        builder.stream(INPUT_TOPIC).to(OUTPUT_TOPIC);
+        final KStream<Long, String> source = builder.stream(INPUT_TOPIC_MAP, Consumed.with(longSerde, stringSerde));
+        final KStream<String, Long> mapped = source.map((key, value) -> new KeyValue<>(value, key));
+        mapped.to(OUTPUT_TOPIC_MAP, Produced.with(stringSerde, longSerde));
+        testDriver = new TopologyTestDriver(builder.build(), config);
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            testDriver.close();
+        } catch (final RuntimeException e) {
+            // https://issues.apache.org/jira/browse/KAFKA-6647 causes exception when executed in Windows, ignoring it
+            // Logged stacktrace cannot be avoided
+            System.out.println("Ignoring exception, test failing in Windows due this exception:" + e.getLocalizedMessage());
```

Comment:
Probably should use a (slf4j) Logger instead of stdout here.

Reply:
Dobe